internal/databroker/config_source.go

```diff
@@ -74,108 +88,131 @@ func (src *ConfigSource) rebuild(ctx context.Context, firstTime firstTime) {
 	_, span := trace.StartSpan(ctx, "databroker.config_source.rebuild")
 	defer span.End()
 
-	log.Info(ctx).Msg("databroker: rebuilding configuration")
-
+	now := time.Now()
 	src.mu.Lock()
 	defer src.mu.Unlock()
+	log.Info(ctx).Str("lock-wait", time.Since(now).String()).Msg("databroker: rebuilding configuration")
 
 	cfg := src.underlyingConfig.Clone()
 
 	// start the updater
 	src.runUpdater(cfg)
 
-	seen := map[uint64]string{}
-	for _, policy := range cfg.Options.GetAllPolicies() {
-		id, err := policy.RouteID()
+	now = time.Now()
+	err := src.buildNewConfigLocked(ctx, cfg)
+	if err != nil {
+		log.Error(ctx).Err(err).Msg("databroker: failed to build new config")
+		return
+	}
+	log.Info(ctx).Str("elapsed", time.Since(now).String()).Msg("databroker: built new config")
+
+	src.computedConfig = cfg
+	if !firstTime {
+		src.Trigger(ctx, cfg)
+	}
+
+	metrics.SetConfigInfo(ctx, cfg.Options.Services, "databroker", cfg.Checksum(), true)
+}
+
+func (src *ConfigSource) buildNewConfigLocked(ctx context.Context, cfg *config.Config) error {
+	eg, ctx := errgroup.WithContext(ctx)
+	eg.SetLimit(runtime.NumCPU()/2 + 1)
+	eg.Go(func() error {
+		src.applySettingsLocked(ctx, cfg)
+		err := cfg.Options.Validate()
 		if err != nil {
-			log.Warn(ctx).Err(err).
-				Str("policy", policy.String()).
-				Msg("databroker: invalid policy config, ignoring")
-			return
+			return fmt.Errorf("validating settings: %w", err)
 		}
-		seen[id] = ""
+		return nil
+	})
+	policies := slices.NewSafeSlice[*config.Policy]()
+	for _, cfgpb := range src.dbConfigs {
+		for _, routepb := range cfgpb.GetRoutes() {
+			routepb := routepb
+			eg.Go(func() error {
```

Comment:
I think this will change the order of routes, which is a breaking change.

Reply:
added sorting